codegen/src/generator/mod.rs

```diff
@@ -84,14 +83,6 @@ fn generate_client<P>(service: &Service, protocol_generator: &P) -> String
             dispatcher: D,
         }}
 
-        impl<P> {type_name}<P, Client> where P: ProvideAwsCredentials {{
```

Comment:
If we're getting rid of the `new` method for services in favor of the method currently named `with_request_dispatcher`, we should rename `with_request_dispatcher` to `new`.  It takes several other arguments, and I only named it that because Rust doesn't support function name overloading.  To the end user, the change will be slightly less jarring, as they'll only have to add an argument to the function they're already calling to get an instance of the service.